crates/optimism/primitives/src/op_tx_type.rs

```diff
@@ -0,0 +1,190 @@
+//! newtype pattern on op_alloy_consensus::OpTxType.
+//! OpTxType that implements reth_primitives_traits::TxType.
+//! This type is required because a Compact impl is needed on the deposit tx type.
+
+use alloy_primitives::{U64, U8};
+use alloy_rlp::{Decodable, Encodable, Error};
+use bytes::BufMut;
+use core::fmt::Debug;
+use derive_more::{
+    derive::{From, Into},
+    Display,
+};
+use op_alloy_consensus::OpTxType as AlloyOpTxType;
+use std::convert::TryFrom;
+
+/// Wrapper type for `AlloyOpTxType` to implement `TxType` trait.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Display, Ord, Hash, From, Into)]
+#[from(AlloyOpTxType)]
```

Comment:
believe this attribute is redundant, the compiler will infer it

Reply:
done! thanks!